contract/test/test-sell-concert-tickets-contract.js

```diff
@@ -80,40 +103,55 @@ test('Start the contract', async t => {
  * @param {ZoeService} zoe
  * @param {ERef<import('@agoric/zoe/src/zoeService/utils').Instance<AssetContractFn>} instance
  * @param {Purse} purse
- * @param {string[]} choices
+ * @param {[string, NatValue][]} choices
+ * @param {boolean} successfulTrade
  */
-const alice = async (t, zoe, instance, purse, choices = ['map', 'scroll']) => {
+const alice = async (
+  t,
+  zoe,
+  instance,
+  purse,
+  choices = [
+    ['frontRow', 1n],
+    ['middleRow', 1n],
+  ],
+  successfulTrade = true,
+) => {
   const publicFacet = E(zoe).getPublicFacet(instance);
   // @ts-expect-error Promise<Instance> seems to work
   const terms = await E(zoe).getTerms(instance);
-  const { issuers, brands, tradePrice } = terms;
+  const { issuers, brands } = terms;
 
-  const choiceBag = makeCopyBag(choices.map(name => [name, 1n]));
+  const choiceBag = makeCopyBag(choices);
+  const totalPrice = bagPrice(choiceBag, terms.inventory);
   const proposal = {
-    give: { Price: tradePrice },
-    want: { Items: AmountMath.make(brands.Item, choiceBag) },
+    give: { Price: totalPrice },
+    want: { Tickets: AmountMath.make(brands.Ticket, choiceBag) },
   };
-  const pmt = await E(purse).withdraw(tradePrice);
+  const pmt = await E(purse).withdraw(totalPrice);
   t.log('Alice gives', proposal.give);
-  // #endregion makeProposal
 
   const toTrade = E(publicFacet).makeTradeInvitation();
 
   const seat = E(zoe).offer(toTrade, proposal, { Price: pmt });
-  const items = await E(seat).getPayout('Items');
+  const tickets = await E(seat).getPayout('Tickets');
 
-  const actual = await E(issuers.Item).getAmountOf(items);
+  const actual = await E(issuers.Ticket).getAmountOf(tickets);
   t.log('Alice payout brand', actual.brand);
   t.log('Alice payout value', actual.value);
-  t.deepEqual(actual, proposal.want.Items);
+  if (successfulTrade) {
+    t.deepEqual(actual, proposal.want.Tickets);
+  } else {
+    t.deepEqual(actual, AmountMath.makeEmptyFromAmount(actual));
```

Comment:
_question_

I'm also getting a unhandled rejection in this case, not sure what's the pattern in ava to assert "I expect this rejection to happen in this case".

Error:
```
Unhandled rejection in test/test-sell-concert-tickets-contract.js

  Error: Not enough inventory, {"brand":"[Alleged: Ticket brand]","value":"[copyBag]"} wanted
    at tradeHandler (.../agoric-basics-contract/src/sell-concert-tickets.contract.js:182:6)
    at eval (.../pass-style/src/make-far.js:161:31)

  › tradeHandler (.../agoric-basics-contract/src/sell-concert-tickets.contract.js:182:6)
  › eval (.../pass-style/src/make-far.js:161:31)
```

Reply:
:thinking: ...

Oh! Yeah... this is definitely an awkward part of the Zoe API.
We forgot to call `E(seat).getOfferResult()`.
(looks like I did that in https://docs.agoric.com/guides/zoe/#trading-with-offer-safety too.)

I just filed an issue for this.
 - https://github.com/Agoric/agoric-sdk/issues/8939